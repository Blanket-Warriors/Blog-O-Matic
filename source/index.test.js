const path = require("path")
const fs = require("fs")


test("Should create content.json", () => {
  return require("./index").buildBlog.then(outputPath => {
    const output = path.join(outputPath, "./content.json")
    const contentJSON = JSON.parse(fs.readFileSync(output, "utf8"))
    expect(contentJSON).toBeTruthy()
  })
})


test("Should create metadata.json", () => {
  return require("./index").buildBlog.then(outputPath => {
    const output = path.join(outputPath, "./posts/coffee/flywheel/metadata.json")
    const contentJSON = JSON.parse(fs.readFileSync(output, "utf8"))
    expect(contentJSON).toMatchSnapshot()
  })
})


test("Should create photos", () => {
  return require("./index").buildBlog.then(outputPath => {
    [
      "large/IMG_0371.jpg",
      "medium/IMG_0371.jpg",
      "small/IMG_0371.jpg",
      "tiny/IMG_0371.jpg"
    ].forEach(image => {
      expect(fs.existsSync(path.join(outputPath, "./galleries/sausilito", image)))
    })
  })
})
