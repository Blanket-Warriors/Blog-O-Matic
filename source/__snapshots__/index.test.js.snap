// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should create content.json 1`] = `
Object {
  "authors": Array [
    "Ben Pevsner",
  ],
  "metadata": Array [
    Object {
      "author": "Ben Pevsner",
      "contentRoot": "/galleries/sausilito/metadata.json",
      "contentType": "gallery",
      "createdDate": "2017-09-09",
      "id": "Sausilito",
      "tags": Array [
        "photos",
        "travel",
      ],
      "title": "Sausilito Photos",
    },
    Object {
      "author": "Ben Pevsner",
      "contentRoot": "/posts/coffee/flywheel/metadata.json",
      "contentType": "article",
      "createdDate": "2018-05-08",
      "id": "flywheel",
      "series": "San Francisco Coffee",
      "tags": Array [
        "coffee",
      ],
      "title": "Flywheel Coffee",
    },
  ],
  "series": Array [
    "San Francisco Coffee",
  ],
  "tags": Array [
    "photos",
    "travel",
    "coffee",
  ],
}
`;

exports[`Should create metadata.json 1`] = `
Object {
  "author": "Ben Pevsner",
  "contentType": "article",
  "createdDate": "2018-05-08",
  "id": "flywheel",
  "series": "San Francisco Coffee",
  "tags": Array [
    "coffee",
  ],
  "title": "Flywheel Coffee",
  "updatedDate": "2018-05-06",
}
`;
